pkg/cli/image/extract/extract.go

```diff
@@ -593,22 +596,32 @@ func (n *copyFromDirectory) Alter(hdr *tar.Header) (bool, error) {
 type copyFromPattern struct {
 	Base string
 	Name string
+	RootDir bool
 }
 
-func newCopyFromPattern(dir, name string) archive.AlterHeader {
+func newCopyFromPattern(dir, name string, rootDir bool) archive.AlterHeader {
+	if rootDir {
+		return &copyFromPattern{Name: name, RootDir: true}
+	}
 	if !strings.HasSuffix(dir, "/") {
 		dir = dir + "/"
 	}
-	return &copyFromPattern{Base: dir, Name: name}
+	return &copyFromPattern{Base: dir, Name: name, RootDir: false}
 }
 
 func (n *copyFromPattern) Alter(hdr *tar.Header) (bool, error) {
-	if !changeTarEntryParent(hdr, n.Base) {
-		return false, nil
-	}
 	matchName := hdr.Name
```

Comment:
this is my bad, it was an error in the suggestion I gave you, this line `matchName := hdr.Name` is in the wrong place.
You'll have to remove it er just initialize it here like `var matchName string` then, below, add the assignment.

Reply:
Thought that := would initialize the variable into whatever type hdr.Name is. So I didn't do it. But I can add the Following:

```
var matchName string
matchName = hdr.Name
```